src/spikeinterface/postprocessing/amplitude_scalings.py

```diff
@@ -291,10 +365,325 @@ def _amplitude_scalings_chunk(segment_index, start_frame, end_frame, worker_ctx)
             else:
                 local_waveform = traces_with_margin[cut_out_start:cut_out_end, sparse_indices]
             assert template.shape == local_waveform.shape
-            local_waveforms.append(local_waveform)
-            templates.append(template)
+
             linregress_res = linregress(template.flatten(), local_waveform.flatten())
-            scalings.append(linregress_res[0])
-    scalings = np.array(scalings)
+            scalings[spike_index] = linregress_res[0]
+
+        # deal with collisions
+        if len(collisions) > 0:
+            num_spikes_in_previous_segments = int(
+                np.sum([len(spikes[segment_slices[s]]) for s in range(segment_index)])
+            )
+            for spike_index, collision in collisions.items():
+                scaled_amps = fit_collision(
+                    collision,
+                    traces_with_margin,
+                    start_frame,
+                    end_frame,
+                    left,
+                    right,
+                    nbefore,
+                    all_templates,
+                    unit_inds_to_channel_indices,
+                    cut_out_before,
+                    cut_out_after,
+                )
+                # the scaling for the current spike is at index 0
+                scalings[spike_index] = scaled_amps[0]
+
+                # make collision_dict indices "absolute" by adding i0 and the cumulative number of spikes in previous segments
```

Comment:
Not sure that `collision_dict` is the correct naming because `collisions` is the same but with local index no ?

Reply:
yes. I'll call it local VS global